File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int solution(int n) {
    int answer = 0;
    
    string strThird("");
    while(n > 0){
        int tmp = n % 3;
        n /= 3;
        strThird += to_string(tmp);
    }
    
    int len = strThird.size() - 1;
    for(int i = 0; i <= len ; i++){
        int tmp = strThird[i] - '0';
        answer += tmp * pow(3 , len - i);
    }
    
    return answer;
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
class Solution {
    public int solution(int n) {
          int answer = 0;
        
        StringBuilder builder = new StringBuilder();
        
        int num = n;
        while(num > 0){
            builder.append( (num % 3) );
            num = num / 3;
            
        }
        
        String str = builder.toString();
        int index = str.length() - 1;
        for(char c : str.toCharArray()){
            answer += ( (c - '0') * (int) Math.pow(3, index--) );
        }        
        return answer;
    }
}